src/braket/circuits/gates.py

```diff
@@ -215,6 +215,121 @@ def i(
+        return "gphase"
+
+    def adjoint(self) -> list[Gate]:
+        return [GPhase(-self.angle)]
+
+    def to_matrix(self) -> np.ndarray:
+        r"""
+        Generate a 1x1 matrix containing the global phase.
+
+        Unitary matrix:
+            .. math:: \mathtt{gphase}(\gamma) = e^(i \gamma) I_m.
+
+        Returns:
+            np.ndarray: GPhase Matrix
+        """
+        return np.exp(1j * self.angle) * np.eye(1, dtype=complex)
+
+    def bind_values(self, **kwargs) -> AngledGate:
+        return get_angle(self, **kwargs)
+
+    @staticmethod
+    def fixed_qubit_count() -> int:
+        return 0
+
+    @staticmethod
+    @circuit.subroutine(register=True)
+    def gphase(
+        angle: Union[FreeParameterExpression, float],
+        *,
+        control: Optional[QubitSetInput] = None,
+        control_state: Optional[BasisStateInput] = None,
+        power: float = 1,
+    ) -> Instruction | Iterable[Instruction]:
+        """Registers this function into the circuit class.
+
+        Args:
+            angle (Union[FreeParameterExpression, float]): Phase in radians.
+            control (Optional[QubitSetInput]): Control qubit(s). Default None.
+            control_state (Optional[BasisStateInput]): Quantum state on which to control the
+                operation. Must be a binary sequence of same length as number of qubits in
+                `control`. Will be ignored if `control` is not present. May be represented as a
+                string, list, or int. For example "0101", [0, 1, 0, 1], 5 all represent
+                controlling on qubits 0 and 2 being in the \\|0⟩ state and qubits 1 and 3 being
+                in the \\|1⟩ state. Default "1" * len(control).
+            power (float): Integer or fractional power to raise the gate to. Negative
+                powers will be split into an inverse, accompanied by the positive power.
+                Default 1.
+
+        Returns:
+            Instruction | Iterable[Instruction]: GPhase instruction.
+
+        Examples:
+            >>> circ = Circuit().gphase(0.45)
+        """
+        if control is not None:
+            control_qubits = QubitSet(control)
+
+            if control_state is None:
+                control_state = len(control_qubits)
+            control_basis_state = BasisState(control_state, len(control_qubits))
+
+            if not any(control_basis_state):
+                return [
+                    X.x(control_qubits[-1]),
+                    PhaseShift.phaseshift(
+                        control_qubits[-1],
+                        angle,
+                        control=control_qubits[:-1],
+                        control_state=control_basis_state[:-1],
+                        power=power,
+                    ),
+                    X.x(control_qubits[-1]),
+                ]
+
+            rightmost_control_qubit_index = (
+                len(control_qubits) - control_basis_state[::-1].index(1) - 1
+            )
```

Comment:
I still don't agree with this interpretation. I don't think we should be filtering through the control state to find a positive control. The qubit used here should simply be the rightmost.

Reply:
I also don't understand why this is necessary - I also thought we should always be acting on the rightmost qubit as the target. Unless I'm misunderstanding what `negctrl @ gphase(a)` means.